assets/js/modules/analytics-4/datastore/api.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * `modules/analytics-4` data store: api.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+
+const baseInitialState = { };
+const baseActions = {};
+const baseControls = {};
+const baseReducer = ( state, { type } ) => {
+	switch ( type ) {
+		default: {
+			return state;
+		}
+	}
+};
+const baseResolvers = {};
+const baseSelectors = {
+	/**
+	 * Checks if the Admin API is working.
+	 *
+	 * Returns false if there are any errors for getProperties or getWebDataStreams.
+	 *
+	 * Returns true if getProperties or getWebDataStreams has fetched successfully.
+	 *
+	 * Else returns undefined.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(boolean | undefined)} True, false, or undefined.
+	 */
+	isAdminAPIWorking( state ) {
+		for ( const errorKey in state.errors ) {
+			if (
+				errorKey.startsWith( 'getProperties' ) ||
+				errorKey.startsWith( 'getWebDataStreams' )
+			) {
+				return false;
+			}
+		}
+
+		for ( const propertyList of Object.values( state.properties ) ) {
+			if ( propertyList && propertyList.length && propertyList.length > 0 ) {
+				for ( const webdatastreamList of Object.values( state.webdatastreams ) ) {
+					if ( webdatastreamList && webdatastreamList.length && webdatastreamList.length > 0 ) {
+						return true;
+					}
+				}
+			}
+		}
```

Comment:
Because this is a temporary, admittedly kinda hacky selector I think this approach is okay (I approved it in the IB 😅), but this kind of direct state access is generally not something we use or endorse—usually we build selectors, even ones marked as `@private`, for everything, so if the internal shape of our state data changes we don't get bugs.

It might be good to explicitly highlight the approach here as against normal conventions so it's acknowledged and doesn't confuse anyone in the future. Just a comment mentioning that this only exists because we're working with an Alpha, unstable API at the time of development.

Reply:
The thought did cross my mind @tofumatt but I was just following the IB! :angel: 
(Although the IB says to access the properties like `state['properties']` but I thought this was not needed :wink: 

I'll add a comment now